src/together/resources/finetune.py

```diff
@@ -366,11 +379,65 @@ def list_events(self, id: str) -> FinetuneListEvents:
             ),
             stream=False,
         )
-
         assert isinstance(response, TogetherResponse)
 
         return FinetuneListEvents(**response.data)
 
+    def list_checkpoints(self, id: str) -> FinetuneCheckpointList:
+        """
+        List available checkpoints for a fine-tuning job
+
+        Args:
+            id (str): Unique identifier of the fine-tune job to list checkpoints for
+
+        Returns:
+            FinetuneCheckpointList: Object containing list of available checkpoints
+        """
+        events = self.list_events(id).data or []
+
+        checkpoints: List[FinetuneCheckpoint] = []
+
+        for event in events:
+            event_type = event.type
+
+            if event_type == FinetuneEventType.CHECKPOINT_SAVE:
+                formatted_time = format_event_timestamp(event)
+                step = get_event_step(event)
+                checkpoint_name = f"{id}:{step}" if step else id
+
+                checkpoints.append(
+                    FinetuneCheckpoint(
+                        type="Intermediate",
+                        timestamp=formatted_time,
+                        name=checkpoint_name,
+                    )
+                )
+            elif event_type == FinetuneEventType.JOB_COMPLETE:
+                formatted_time = format_event_timestamp(event)
+                is_lora = hasattr(event, "adapter_path")
+
+                checkpoints.append(
```

Comment:
There are some models for which we don't upload the merged checkpoint when training with LoRA. Will the current logic handle them correctly?

Reply:
Changed, this way it will work correctly